tests/unit/test_util.py

```diff
@@ -88,6 +93,33 @@ def test_git_commit_info(self):
         self.assertIsInstance(git_commit_info, str)
         self.assertEqual(git_commit_info[:2], "v2")
 
+    def test_have_optional_dependency(self):
```

Comment:
For the coverage missing, you might want to call the `have_optional_dependency` method directly.

Reply:
Added a case to potentially fix the coverage with direct usage of this in [L-122](https://github.com/pybamm-team/PyBaMM/pull/3475/files#diff-213a36200d41e6897aaba82ed856a037dcf622587acf920011b95c64d3428338R122), thanks.